plugins/hls-rename-plugin/src/Ide/Plugin/Rename.hs

```diff
@@ -48,7 +48,7 @@ renameProvider state pluginId (RenameParams (TextDocumentIdentifier uri) pos _pr
             getFileEdits = ap (getSrcEdits state . renameModRefs newNameText) (locToUri . head)
 
         fileEdits <- mapM getFileEdits filesRefs
-        pure $ foldl1 (<>) fileEdits
+        pure $ foldl' (<>) mempty fileEdits
```

Comment:
If you want a strict version - `foldl1'` is also an option in `base`.

*afaik* `foldl1` is not completely strict, while `foldl1'` is "properly" strict, a submitter of it had a coverage on that & that is pointed out in https://hackage.haskell.org/package/base-4.16.0.0/docs/Data-List.html#v:foldl1-39-, after its submission a lof of data types shipped `foldl1'.

Reply:
TBH, I don't know what level of strictness is needed here. I guess "the more the better" since it is left fold and non-strict left fold does not have much application.

The goal here was to get rid of `foldl1` as it is unsafe. It wasn't obvious for me that the `fileEdits` is a nonempty list. However, since we have monoid here, we can use ordinary `foldl(')` with `mempty` as initial condition.

Reply:
Actually, I think there's a small policy question here: do we want to return a response with an empty edit here, or fail the response? I think either is arguable, but perhaps worth a note that we might send back an empty edit here.

Reply:
As a matter of policy, we don't want any unhandled exceptions to leak out. So we want `fold'` here, and the empty set of edits is fine.

Reply:
I didn't mean to leak an exception, the request itself can fail in the protocol. Probably that's not appropriate since nothing went *wrong* per se, though.